src/main/groovy/com/calgaryscientific/gradle/VeracodeTask.groovy

```diff
@@ -44,50 +44,45 @@ abstract class VeracodeTask extends DefaultTask {
             'fileId'           : '123',
             'mode'             : 'action|actionSummary|verbose',
             'maxUploadAttempts': '123',
-            'fileId'           : 'xxx',
             'buildId1'         : '123',
             'buildId2'         : '123'
     ]
 
-    def requiredArguments = []
+    List<String> requiredArguments = []
+    List<String> optionalArguments = []
 
     VeracodeTask() {
         group = 'Veracode'
     }
 
     abstract void run()
 
-    final String correctUsage() {
-        StringBuilder sb = new StringBuilder("Example of usage: gradle ${getName()}")
-        requiredArguments.each() {
-            if (!isArgumentOptional(it)) {
-                sb.append(" -P${it}=${validArguments.get(it)}")
-            } else {
-                String originalArgument = it.substring(0, it.length() - OPTIONAL.length())
-                sb.append(" [-P${originalArgument}=${validArguments.get(originalArgument)}]")
-            }
+    protected static String correctUsage(String taskName,
```

Comment:
you might want to use some standard command line parse options instead of rolling our own

Reply:
Do you have any sources? I just improved on the existing code base by not adding a suffix to the property and then removing it but by having two separate lists for it.